src/Groups/directproducts.jl

```diff
@@ -317,7 +331,8 @@ end
 
 # start part on subgroups
 function _as_subgroup_bare(G::SemidirectProductGroup{S,T}, H::GapObj) where { S , T }
-  return SemidirectProductGroup(H, G.N, G.H, G.f, G.X, false)
+  t = G.X==H
```

Comment:
As I explained before: this is a potentially expensive (even infeasible!) computation. I think the whole concept of storing the "fullness" information this way is flawed. At the very least, it should be stored with three possible values: true, false and unknowon.

Reply:
At the moment I put `false` as default argument. Instead of `G.X==H`, could be reasonably faster something like checking whether the index of `H` in `G.X` is 1, or check whether the generators of `G.X` lie in `H`?